### ISSUE TITLE
GATConv and GATv2Conv attending to all other nodes

### ISSUE THREAD

**Issue by mahadafzal** [rusty1s/pytorch_geometric#3057]
The paper and the documentation provided on the landing page state that node i attends to all node j's where j nodes are in the neighborhood of i. Is there a way to go back to GAT's general formulation where node i attends to **all** other nodes, not just its neighborhood? This is mentioned under Section 2.1 of [Graph Attention Networks](https://arxiv.org/pdf/1710.10903.pdf).

**Comment by rusty1s** [comment]
Yes, this is possibly by letting `edge_index` denote the fully-connected graph, e.g.:
```python
num_nodes = 10
row = torch.arange(num_nodes).view(-1, 1).repeat(1, num_nodes).view(-1)
col = torch.arange(num_nodes).view(-1, 1).repeat(num_nodes, 1).view(-1)
edge_index = torch.stack([row, col], dim=0)
```
However, PyG is optimized for sparse-structures, and there exist more efficient implementations for `GATConv`, e.g., Transformers, when operating on fully-connected graphs.
